File: examples/tax.py
from selenium.webdriver.common.by import By
def tax_myql_oracle(driver):
    state = driver.execute_script('return $state;');
    
    assert not True or driver.find_elements(By.ID, 'hasBoughtHouse_div_31')[0].is_displayed(), 'true evaluated to true, however hasBoughtHouse_div_31 was not displayed'
    assert not True or driver.find_elements(By.ID, 'hasMaintLoan_div_95')[0].is_displayed(), 'true evaluated to true, however hasMaintLoan_div_95 was not displayed'
    assert not True or driver.find_elements(By.ID, 'hasSoldHouse_div_152')[0].is_displayed(), 'true evaluated to true, however hasSoldHouse_div_152 was not displayed'
    assert not (True and state['hasSoldHouse']) or driver.find_elements(By.ID, 'sellingPrice_div_244')[0].is_displayed(), 'true && hasSoldHouse evaluated to true, however sellingPrice_div_244 was not displayed'
    assert not (True and state['hasSoldHouse']) or driver.find_elements(By.ID, 'privateDebt_div_306')[0].is_displayed(), 'true && hasSoldHouse evaluated to true, however privateDebt_div_306 was not displayed'
    assert not (True and state['hasSoldHouse']) or driver.find_elements(By.ID, 'valueResidue_div_373')[0].is_displayed(), 'true && hasSoldHouse evaluated to true, however valueResidue_div_373 was not displayed'
    assert state['valueResidue'] == (state['sellingPrice'] - state['privateDebt']), ' valueResidue was not the same as sellingPrice - privateDebt'

File: examples/tax.js
var $state = {
  
  hasMaintLoan: false,
  
  hasSoldHouse: false,
  
  privateDebt: 0,
  
  sellingPrice: 0,
  
  valueResidue: 0,
  
  hasBoughtHouse: false,
  
}
function $update(name, value) {
   let change = false;
   let newVal = undefined;
   let div = undefined;
   if (name !== undefined) {
      $state[name] = value;
   }
   do {
     change = false;
     
     div = document.getElementById('hasBoughtHouse_div_31');
     div.style.display = true ? 'block' : 'none'; 
     
     
     div = document.getElementById('hasMaintLoan_div_95');
     div.style.display = true ? 'block' : 'none'; 
     
     
     div = document.getElementById('hasSoldHouse_div_152');
     div.style.display = true ? 'block' : 'none'; 
     
     
     div = document.getElementById('sellingPrice_div_244');
     div.style.display = (true && $state.hasSoldHouse) ? 'block' : 'none'; 
     
     
     div = document.getElementById('privateDebt_div_306');
     div.style.display = (true && $state.hasSoldHouse) ? 'block' : 'none'; 
     
     
     div = document.getElementById('valueResidue_div_373');
     div.style.display = (true && $state.hasSoldHouse) ? 'block' : 'none'; 
     
     newVal = ($state.sellingPrice - $state.privateDebt);
     if (newVal !== $state.valueResidue) {
       let elt = document.getElementById('valueResidue_widget_373');
       $state.valueResidue = newVal;
       
       elt.value = newVal;
       
       change = true;
     }
     
     
   } while (change);
}
